includes/class-wc-payments.php

```diff
@@ -935,6 +935,22 @@ function () {
 					}
 				}
 			);
+
+			// This injects the payments API into core, so the WooCommerce Blocks plugin is not necessary.
+			// The payments API is currently only available in feature builds (with flag `WC_BLOCKS_IS_FEATURE_PLUGIN`).
+			// We should remove this once it's available in core by default.
+			if ( ! defined( 'WC_BLOCKS_IS_FEATURE_PLUGIN' ) && class_exists( 'Automattic\WooCommerce\Blocks\Payments\Api' ) ) {
```

Comment:
As discussed over the [thread](p1650642013069719/1650319227.659789-slack-C022WMN88KG) let's add a condition here to check if `woopay` is eligible and enabled.

Reply:
We're doing it right above on line 924 🙂.

Reply:
My bad. It seemed to me outside of that `if` check :facepalm: 
Sorry for the confusions :facepalm:

Reply:
It's easy to miss that since GH doesn't display the full scope of where the diff is 🙂.